client/asset/btc/btc.go

```diff
@@ -824,25 +824,67 @@ func (btc *ExchangeWallet) FundingCoins(ids []dex.Bytes) (asset.Coins, error) {
 			coins = append(coins, newOutput(btc.node, txHash, vout, fundingCoin.amount, fundingCoin.redeemScript))
 			continue
 		}
-		notFound[pt] = struct{}{}
+		notFound[pt] = true
 	}
-
 	if len(notFound) == 0 {
 		return coins, nil
 	}
+
+	// Check locked outputs for not found coins.
+	lockedOutpoints, err := btc.wallet.ListLockUnspent()
+	if err != nil {
+		return nil, err
+	}
+	for _, rpcOP := range lockedOutpoints {
+		txHash, err := chainhash.NewHashFromStr(rpcOP.TxID)
+		if err != nil {
+			return nil, fmt.Errorf("error decoding txid from rpc server %s: %v", rpcOP.TxID, err)
+		}
+		pt := newOutPoint(txHash, rpcOP.Vout)
+		if notFound[pt] {
+			txOut, err := btc.node.GetTxOut(txHash, rpcOP.Vout, true)
+			if err != nil {
+				return nil, fmt.Errorf("gettxout error for locked outpoint %v: %v", pt.String(), err)
+			}
+			var address string
+			if len(txOut.ScriptPubKey.Addresses) > 0 {
+				address = txOut.ScriptPubKey.Addresses[0]
+			}
+			pkScript, err := hex.DecodeString(txOut.ScriptPubKey.Hex)
+			if err != nil {
+				return nil, fmt.Errorf("error decoding pubkey script from hex '%s': %v",
+					txOut.ScriptPubKey.Hex, err)
+			}
+			nfo, err := dexbtc.InputInfo(pkScript, nil, btc.chainParams)
+			if err != nil {
+				return nil, fmt.Errorf("error reading asset info: %v", err)
+			}
+			utxo := &compositeUTXO{
+				txHash:       txHash,
+				vout:         rpcOP.Vout,
+				address:      address,
+				redeemScript: nil,
+				amount:       toSatoshi(txOut.Value),
+				input:        nfo,
+			}
+			coin := newOutput(btc.node, txHash, rpcOP.Vout, toSatoshi(txOut.Value), nil)
+			coins = append(coins, coin)
+			btc.fundingCoins[pt] = utxo
+			delete(notFound, pt)
+			if len(notFound) == 0 {
+				return coins, nil
+			}
+		}
+	}
+
```

Comment:
Should throw a 

```
if len(notFound) == 0 {
	return coins, nil
}
```
here.

Reply:
Ah, damn. Came to mind while working but forget to slip it in before committing. Thanks!

Reply:
Err, looks like that may not be necessary.
We already know that `len(notFound) > 0` before fetching and looping through locked outpoints.
The only way that would become 0 is if the loop deletes from the notFound map, and there's a 0-check in the loop when that is done.